SPDX-FileCopyrightText: 2025 fosslinux <fosslinux@aussies.space>
SPDX-FileCopyrightText: 2012 Karl Williamson <public@khwilliamson.com>

SPDX-License-Identifier: Artistic-1.0

From 914c32b1b85e310c832192ef133d2eb8b7108bfa Mon Sep 17 00:00:00 2001
From: Samuel Tyler <fosslinux@aussies.space>
Date: Fri, 22 Aug 2025 23:38:59 +1000
Subject: [PATCH] Revert "regen/regcharclass.pl: Generate macros for \X
 processing"

The _X_* family of symbols were added in the previous commit, so we cannot
use them this build. The next two patches in the series removes the use of the
macros generated by regcharclass.pl.

This reverts commit 612ead590b8b5f05e4060738540192ece946c340.
---
 regen/regcharclass.pl | 28 ----------------------------
 1 file changed, 28 deletions(-)

diff --git perl-5.17.4/regen/regcharclass.pl perl-5.17.4/regen/regcharclass.pl
index 7d126428ef..e11b97003e 100755
--- perl-5.17.4/regen/regcharclass.pl
+++ perl-5.17.4/regen/regcharclass.pl
@@ -1133,34 +1133,6 @@ SURROGATE: Surrogate characters
 => UTF8 :fast
 \p{Gc=Cs}
 
-GCB_L: Grapheme_Cluster_Break=L
-=> UTF8 :fast
-\p{_X_GCB_L}
-
-GCB_LV_LVT_V: Grapheme_Cluster_Break=(LV or LVT or V)
-=> UTF8 :fast
-\p{_X_LV_LVT_V}
-
-GCB_Prepend: Grapheme_Cluster_Break=Prepend
-=> UTF8 :fast
-\p{_X_GCB_Prepend}
-
-GCB_RI: Grapheme_Cluster_Break=RI
-=> UTF8 :fast
-\p{_X_RI}
-
-GCB_SPECIAL_BEGIN: Grapheme_Cluster_Break=special_begins
-=> UTF8 :fast
-\p{_X_Special_Begin}
-
-GCB_T: Grapheme_Cluster_Break=T
-=> UTF8 :fast
-\p{_X_GCB_T}
-
-GCB_V: Grapheme_Cluster_Break=V
-=> UTF8 :fast
-\p{_X_GCB_V}
-
 # This program was run with this enabled, and the results copied to utf8.h;
 # then this was commented out because it takes so long to figure out these 2
 # million code points.  The results would not change unless utf8.h decides it
-- 
2.49.1

